### DISCUSSION TITLE
cdc 任务数无法超过 20 个

### DISCUSSION THREAD

**Discussion by tan-zhuo** [apache/flink-cdc#2242] [Q&A]
本地部署了 2个 taskManger 节点，1个 jobManager节点  其中 1个taskManager与JobManager部署在同一台服务器上。
我使用CDC进行作业，前面20个作业正常运行。但到了 21 个任务创建时出现了连接超时异常。

![image](https://github.com/ververica/flink-cdc-connectors/assets/32378959/9dc2c9f5-1a48-46c1-9709-dee3770737b3)

![image](https://github.com/ververica/flink-cdc-connectors/assets/32378959/8a3da5f8-824c-4706-9426-ee22b5de8edb)

我查看了官方文档，但依然不确定需要修改什么配置。为什么CDC任务只能创建20个呢？
我确认了我的mysql数据库可用连接数充足，并且taskManger的资源充足。

**Comment by leonardBang** [comment]
mysql cdc 默认的数据库连接池是20， 可以设置这个参数看下

**Reply by leonardBang** [reply]
connection.pool.size
https://ververica.github.io/flink-cdc-connectors/release-2.4/content/connectors/mysql-cdc%28ZH%29.html

**Reply by zjf0128** [reply]
请问你有设置成功过吗，我设置了这个参数也不行。connection.pool.size

**Comment by zjf0128** [comment]
后面有解决没有，我也是这个问题。

**Reply by tan-zhuo** [reply]
不好意思，我也没有解决 😂 。集群里面去配置 connection.pool.size 确实是不生效的

**Comment by lixiaoqiang7** [comment]
也遇到了这个问题，有解决吗

**Reply by zhbdesign** [reply]
好像是伪装从库多了不行，所以datastream多表比较好，群里也见别人遇到这问题，不知道有没有解决

**Reply by ysmintor** [reply]
你说的 datastream 是指用 Flink datastream api 方式处理？
@zhbdesign

**Reply by X1304951263** [reply]
应该是的   datastream  可以  多表 处理，一个任务  几十张表 都没问题，但是需要自己写代码

**Comment by zhanguomozhu** [comment]
有解决方案吗？设置的参数完全没用

**Reply by beryllw** [reply]
connection.pool.size
https://nightlies.apache.org/flink/flink-cdc-docs-release-3.2/docs/connectors/pipeline-connectors/mysql/
![image](https://github.com/user-attachments/assets/12ab60f0-997b-46e0-88e6-c4a1ac07b30e)
设置在Flink集群中，还是cdc pipeline MySQL Source中了？测试一下设置在CDC pipeline MySQL source中。
